### DISCUSSION TITLE
Unable to Compile Desktop project

### DISCUSSION THREAD

**Discussion by PetarSlavov** [unoplatform/uno#7617] [Q&A] [answered]
I have the latest version of uno-check 1.1.0-dev.9 and it does not report any problems
Latest version of Uno project templates 4.0.9. I'm creating a net project with dotnet new unoapp-winui-net6. Then opening the project with the latest version of Visual Studio 2022 (17.0.2) and getting rid of all mobile project heads. Leaving only Shared, Wasm, Windows.Desktop and Windows.Desktop.Shared.
I also have the latest SDKs installed 5.0.403 and 6.0.100.

I'm getting two build error which I don't know how to resolve:

This version of WindowsAppSDK requires WinRT.Runtime.dll version 1.3 or greater.
    Please update to .NET SDK 5.0.205, 5.0.302, or 5.0.400 (or later).

AND

A referenced assembly was compiled using a newer version of Microsoft.Windows.SDK.NET.dll. Please update to a newer .NET SDK in order to reference this assembly.

**Comment by jeromelaban** [comment]
This is related to this issue: https://github.com/microsoft/WindowsAppSDK/issues/745

You may need to update this section to later versions of the packages: 

https://github.com/unoplatform/uno/blob/9fd0a0c5de0f5d07e5657d890484d7621a4aa04e/src/SolutionTemplate/Uno.ProjectTemplates.Dotnet/content/unoapp-winui-net6/UnoWinUIQuickStart.Windows.Desktop/UnoWinUIQuickStart.Windows.Desktop.csproj#L24-L27

**Reply by PetarSlavov** [reply]
Updating this section to 10.0.19041.16 helped get rid off one of the errors. 

I'm still unable to build the project due to the other error:

This version of WindowsAppSDK requires WinRT.Runtime.dll version 1.3 or greater.
Please update to .NET SDK 5.0.205, 5.0.302, or 5.0.400 (or later).

**Reply by jeromelaban** [reply]
The latest version for these packages is `10.0.18362.22`. Is this the one you previously had ?

**Reply by PetarSlavov** [reply]
What I previously had was 10.0.18362.16, changing to  10.0.18362.22 help resolve both errors and the project build successfully, but I had to change the target processor architecture to x86. Any hints at why this is?

**Reply by PetarSlavov** [reply]
Also, when I try to run the Desktop project and get this exception:

Unable to load DLL 'Microsoft.ui.xaml.dll' or one of its dependencies: The specified module could not be found. (0x8007007E)

**Reply by jeromelaban** [reply]
Got it. The desktop project cannot be started, use the Package project instead. See https://platform.uno/docs/articles/get-started-winui3.html#create-a-new-app. The target architecture must be either x86 or x64 in the project configuration to run WinUI projects on windows.

**Reply by PetarSlavov** [reply]
Thank you for your help. I recreated the project, set target architecture to x86 and ran the Package project. It  builds and runs, but I'm still getting an error:

System.EntryPointNotFoundException: 'Unable to find an entry point named 'XamlCheckProcessRequirements' in DLL 'Microsoft.ui.xaml.dll'.'

**Reply by PetarSlavov** [reply]
I was able to resolve this exception by changing this in Package.wapproj
From:
```
<PackageReference Include="Microsoft.WindowsAppSDK" Version="[1.0.0-experimental1]">
```
To:
```
<PackageReference Include="Microsoft.WindowsAppSDK" Version="[1.0.0]">
```
As I only had this version of WindowsAppSDK installed.

**Reply by jeromelaban** [reply]
Thanks, we'll make changes to the template so it works properly.

**Reply by MaiastraMan** [reply]
I have this exact problem: First I got the "Unable to load DLL 'Microsoft.ui.xaml.dll' so I ran the package not the project.
Then I got the "System.EntryPointNotFoundException" so I inspected the Package.wapproj file.
However, I found that I already have the following line in this file: 
`<PackageReference Include="Microsoft.WindowsAppSDK" Version="[1.2.230217.4]">`
Does anyone have any ideas of what else I could try, to fix this?

**Reply by jeromelaban** [reply]
`.wapproj` are not supposed to be used anymore. Which project did you use as base?

**Reply by MaiastraMan** [reply]
I am using the "Blank App, Packaged with Windows Application Packaging Project (WinUI3 in Desktop)" template. If I double click on the package name in solution explorer, the `.wapproj` file opens. I also confirmed that the `.csproj` file contains the above mentioned elements.

**Reply by jeromelaban** [reply]
Thanks. The template you selected comes from the official WinAppSDK, which is different from [the templates coming from Uno](https://platform.uno/docs/articles/get-started-vs-2022.html#install-the-solution-templates). Uno's templates don't have that problem as they're being setup to avoid it.

If you intend to make an Uno app, what made you choose the `Blank App, Packaged with Windows Application Packaging Project (WinUI3 in Desktop)` template, and not the `Uno Platform App` template?

**Reply by MaiastraMan** [reply, marked as answer]
Thank you very much for explaining this to me, to be honest I somehow failed to notice that this repository was for Uno specifically, and my project is not for this platform. Apologies for taking up your time, but just in case it helps anyone else, I managed to fix my problem by removing the following line from my .csproj file: `<PackageReference Include="Microsoft.UI.Xaml" Version="2.8.4" />`
